src/main/java/org/folio/des/service/bursarlegacy/BursarMigrationService.java

```diff
@@ -0,0 +1,448 @@
+import org.folio.des.domain.dto.BursarExportTokenConditionalConditionsInner;
+import org.folio.des.domain.dto.BursarExportTokenConstant;
+import org.folio.des.domain.dto.BursarExportTokenDateType;
+import org.folio.des.domain.dto.BursarExportTokenFeeAmount;
+import org.folio.des.domain.dto.BursarExportTokenFeeDate;
+import org.folio.des.domain.dto.BursarExportTokenFeeMetadata;
+import org.folio.des.domain.dto.BursarExportTokenLengthControl;
+import org.folio.des.domain.dto.BursarExportTokenUserDataOptional;
+import org.folio.des.domain.dto.BursarExportTransferCriteria;
+import org.folio.des.domain.dto.BursarExportTransferCriteriaConditionsInner;
+import org.folio.des.domain.dto.BursarExportTransferCriteriaElse;
+import org.folio.des.domain.dto.ExportTypeSpecificParameters;
+import org.folio.des.domain.dto.Job;
+import org.folio.des.domain.dto.JobWithLegacyBursarParameters;
+import org.folio.des.domain.dto.JobWithLegacyBursarParametersCollection;
+import org.folio.des.domain.dto.LegacyBursarFeeFines;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMapping;
+import org.folio.des.domain.dto.LegacyBursarFeeFinesTypeMappings;
+import org.folio.des.service.JobService;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class BursarMigrationService {
+  private static final Integer DEFAULT_LIMIT = 10000;
+
+  private static final String BURSAR_EXPORT_MIGRATION_HEADER_CONSTANT = "LIB02";
+
+  public boolean isLegacyJob(JobWithLegacyBursarParameters job) {
+    LegacyBursarFeeFines bursarFeeFines = job
+      .getExportTypeSpecificParameters()
+      .getBursarFeeFines();
+
+    return bursarFeeFines.getDaysOutstanding() != null;
+  }
+
+  public void recreateLegacyJobs(
+    BursarExportLegacyJobService bursarExportLegacyJobService,
+    JobService jobService
+  ) {
+    log.info("searching for legacy jobs");
+    JobWithLegacyBursarParametersCollection response = bursarExportLegacyJobService.get(
+      0,
+      DEFAULT_LIMIT,
+      "status==SCHEDULED"
+    );
+    List<JobWithLegacyBursarParameters> jobsToRecreate = new ArrayList<>();
+
+    int total = response.getTotalRecords();
+    jobsToRecreate.addAll(response.getJobRecords());
+    while (jobsToRecreate.size() < total) {
+      response =
+        bursarExportLegacyJobService.get(
+          jobsToRecreate.size(),
+          DEFAULT_LIMIT,
+          "status==SCHEDULED"
+        );
+      jobsToRecreate.addAll(response.getJobRecords());
+    }
+
+    for (JobWithLegacyBursarParameters jobWithLegacyBursarParameters : jobsToRecreate) {
+      log.info("job to recreate: {}", jobWithLegacyBursarParameters);
+      if (isLegacyJob(jobWithLegacyBursarParameters)) {
+        Job newJob = prepareNewJob();
+
+        newJob
+          .getExportTypeSpecificParameters()
+          .setVendorEdiOrdersExportConfig(
```

Comment:
any reasons to touch edi, authrorities etc? as I understand you need only bursar jobs - and in previous implementation it was possible to schedule only one bursar job

Reply:
I mean to have only one config

Reply:
This has been resolved. You can verify the filtering logic in `BursarExportLegacyJobService`